### PR TITLE
Add `git-diff` option to `terrahub run` command

### LINKED ISSUE #294: Add `--git-diff` option to `terrahub run` to optimize / reduce the list of executed components
## Describe the Feature

Add `--git-diff` option to `terrahub run` to optimize / reduce the list of executed components

## Expected Behavior

If option `--git-diff X..Y` (or `-g X..Y`) is passed to `terrahub run`, I expect the following to happen:
1. `git diff --name-only X Y` provides a list of files changed between commit X and commit Y
2. above list is compared to `ci.mapping` definitions across all `.terrahub.yml` config files and matched terrahub components are compiled into a list
3. if list is empty, execute `terrahub run`
4. if list is NOT empty, execute `terrahub run --include [above_list_as_csv]`
NOTE: in theory, a pull request can be viewed as a `git diff` between commit X and commit Y

### PR DESCRIPTION
**Pull request by Leichtwind**
## Description

Fixes #294

## Type of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] This change requires a change to the documentation

## How Has This Been Tested?

1. Create a `terrahub` project
2. Initialise a `git` repository
3. Test different cases of components' and project's `ci mapping` configurations

## Checklist:

- [ ] I have read the **CONTRIBUTING** document.
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have checked that my changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/terraform-command.js [hunk 1]
```diff
@@ -173,14 +186,88 @@ class TerraformCommand extends AbstractCommand {
...
+    return this._configLoader.getProjectCi();
+  }
+
+  /**
+   * @return {String[]}
+   */
+  getGitDiff() {
+    const commits = this.getOption('git-diff');
+
+    if (!commits.length) {
```
Comment (ddimitrioglo) [inline src/terraform-command.js]: this is covered in `.addOption('git-diff', 'g', 'To be updated', Array, [])`, am I wrong?

File: src/terraform-command.js [hunk 2]
```diff
@@ -173,14 +186,88 @@ class TerraformCommand extends AbstractCommand {
...
+    const diffList = stdout.toString().split('\n').slice(0, -1).map(it => join(this.getAppPath(), it));
+
+    if (!diffList.length) {
+      return [];
+    }
+
+    const config = super.getConfig();
+    const projectCiMapping = this.getProjectCi() ? this.getProjectCi().mapping : [];
+
+    const isAll = (projectCiMapping || []).some(dep => this._compareCiMappingToGitDiff(dep, diffList));
```
[the remark this answers is not in the document]
Comment (Leichtwind) [reply src/terraform-command.js]: Yes, it is. I will remove it today

File: src/terraform-command.js (hunk 1 above)
Comment (Leichtwind) [reply src/terraform-command.js]: I don't want to run `git diff` and all the code below, if option `git-diff` is not used

File: src/terraform-command.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (Leichtwind) [reply src/terraform-command.js]: Hmmm, no it isn't
The first line covers case when `ci` is missing in config
The second line covers case when `mapping` is missing in `ci`

**Pull request merged**